hudi-spark-datasource/hudi-spark/src/main/java/org/apache/hudi/bootstrap/SparkFullBootstrapDataProviderBase.java

```diff
@@ -84,7 +84,7 @@ public JavaRDD<HoodieRecord> generateInputRecords(String tableName, String sourc
                   Boolean.parseBoolean(KeyGeneratorOptions.KEYGENERATOR_CONSISTENT_LOGICAL_TIMESTAMP_ENABLED.defaultValue())));
           try {
             return DataSourceUtils.createHoodieRecord(gr, orderingVal, keyGenerator.getKey(gr),
-                props.getString("hoodie.datasource.write.payload.class"));
+                props.getString("hoodie.datasource.write.payload.class"), null);
```

Comment:
why is this change ?

Reply:
We added an extra argument so that `DataSourceUtils.createHoodieRecord` can take `scala.Option<HoodieRecordLocation> recordLocation` as an input argument. Here there is no recordLocation specified so passing in `Option.apply(null)` to maintain original behavior. I could add another signature for `DataSourceUtils.createHoodieRecord` to avoid adding null here?